src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1240,6 +1253,39 @@ - (void)_createDataVersionFilterListFromDictionary:(NSDictionary<MTRClusterPath
     *count = maxDataVersionFilterSize;
 }
 
+- (void)_setupConnectivityMonitoring
+{
+    std::lock_guard lock(_lock);
+
+    if (_connectivityMonitor) {
+        // already monitoring
+        return;
+    }
+
+    // Get the required info before setting up the connectivity monitor
+    NSNumber * compressedFabricID = [_deviceController syncGetCompressedFabricID];
+    if (!compressedFabricID) {
+        MTR_LOG_INFO("%@ could not get compressed fabricID", self);
+        return;
+    }
+
+    char instanceName[chip::Dnssd::kMaxOperationalServiceNameSize];
+    chip::PeerId peerId(static_cast<chip::CompressedFabricId>(compressedFabricID.unsignedLongLongValue), static_cast<chip::NodeId>(_nodeID.unsignedLongLongValue));
+    CHIP_ERROR err = chip::Dnssd::MakeInstanceName(instanceName, sizeof(instanceName), peerId);
+    if (err != CHIP_NO_ERROR) {
+        MTR_LOG_ERROR("%@ could not make instance name", self);
+        return;
+    }
```

Comment:
Feels like this low-level name mangling belongs at least in a helper function, or in the class that actually deals with the DNS-SD mechanics.

Reply:
Moved it into MTRDeviceConnectivityMonitor itself.